### PR TITLE
Add pubsub message validation against JSON schemas

### PR DESCRIPTION
**Pull request by econchick**
**What this PR does / why we need it**:

This is essentially adapting to what @descent3me [originally proposed](https://github.com/spotify/gordon/pull/4/) for core but moved to this plugin as core shouldn't have to worry about specific plugin message validation.

This adds a module/class that will load packaged schemas (as merged by #2 ) and provides a method to validate a message against a loaded schema.

**Which issue(s) this PR fixes** 

N/A

**Special notes for your reviewer**:

Some docs may need to be changed once the plugin logic actually uses `MessageValidator` class. I just made some simple assumptions.

Tests will not look like their passing because of undefined interface scaffolding.

**Release note**:

```release-note
NONE
```

cc @spotify/alf

### CODE REVIEW

File: gordon_gcp/schema/validate.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+To use:
+
+.. code-block:: pycon
+
+    >>> from gordon_gcp.schema import validate
+    >>> validator = validate.MessageValidator()
+    >>> validator.schemas
+    {
+        'event': {'event': 'schema'},
+        'audit-log': {'audit-log': 'schema'}
```
[the remark this answers is not in the document]
Comment (econchick) [reply gordon_gcp/schema/validate.py]: ah yeah that's a fair point, good call.

File: gordon_gcp/exceptions.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from gordon import exceptions as core_exceptions
+
+
+class GCPGordonError(core_exceptions.GordonError):
+    """General Gordon GCP Plugin Error."""
```
[the remark this answers is not in the document]
Comment (econchick) [reply gordon_gcp/exceptions.py]: I'm not sure if this is a statement, a question, or a suggestion...?

[the remark this answers is not in the document]
Comment (descent3me) [reply gordon_gcp/exceptions.py]: Just a statement. I wish other external APIs I've used had done this as well.

File: gordon_gcp/schema/validate.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    def __init__(self):
+        self.schemas = self._load_schemas()
+
+    def _load_schemas(self):
+        schema_path = pathlib.Path(self.HERE, self.SCHEMA_DIR).absolute()
+        schema_path_contents = schema_path.glob('*.json')
+
+        schemas = {}
+        for schema_file in schema_path_contents:
+            schema_name = schema_file.name.split('.')[0]
```
[the remark this answers is not in the document]
Comment (econchick) [reply gordon_gcp/schema/validate.py]: The files are named `filename.schema.json`, so the stem returns `filename.schema`, which isn't as pretty as `filename`.

[the remark this answers is not in the document]
Comment (matt0bi) [reply gordon_gcp/schema/validate.py]: Gotcha :+1:, didn't catch that!

**Comment by econchick** [comment]
@descent3me @matt0bi I've hopefully addressed everything here, mind taking a quick look?

**Comment by descent3me** [comment]
@econchick Almost! Reconciliation is still spelled wrong 😛 . After that, I'm +1

**Comment by matt0bi** [comment]
:+1:

**Pull request merged**